fail2ban/tests/dummyjail.py

```diff
@@ -24,43 +24,66 @@
 
 from threading import Lock
 
+from ..server.jail import Jail
 from ..server.actions import Actions
 
 
-class DummyJail(object):
+class DummyJail(Jail, object):
 	"""A simple 'jail' to suck in all the tickets generated by Filter's
 	"""
-	def __init__(self):
+	def __init__(self, backend=None):
 		self.lock = Lock()
 		self.queue = []
-		self.idle = False
-		self.database = None
-		self.actions = Actions(self)
+		super(DummyJail, self).__init__(name='DummyJail', backend=backend)
+		self.__db = None
+		self.__actions = Actions(self)
 
 	def __len__(self):
-		try:
-			self.lock.acquire()
+		with self.lock:
 			return len(self.queue)
-		finally:
-			self.lock.release()
+
+	def isEmpty(self):
+		with self.lock:
+			return not self.queue
+
+	def isFilled(self):
+		with self.lock:
+			return bool(self.queue)
 
 	def putFailTicket(self, ticket):
-		try:
-			self.lock.acquire()
+		with self.lock:
 			self.queue.append(ticket)
-		finally:
-			self.lock.release()
 
 	def getFailTicket(self):
-		try:
-			self.lock.acquire()
+		with self.lock:
 			try:
 				return self.queue.pop()
 			except IndexError:
 				return False
-		finally:
-			self.lock.release()
 
 	@property
 	def name(self):
 		return "DummyJail #%s with %d tickets" % (id(self), len(self))
+
+	@property
+	def idle(self):
+		return False;
+	
+	@idle.setter
+	def idle(self, value):
+		pass
```

Comment:
or if intentionally need to be ignored -- please add a comment on that (also why needs to be ignored in tests)